redash/query_runner/activedata.py

```diff
@@ -0,0 +1,180 @@
+                schema[table_name] = {'name': table_name, 'columns': []}
+
+            schema[table_name]['columns'].append(row['name'])
+
+        return [{'name': r['name'], 'columns': sorted(r['columns'])} for r in schema.values()]
+
+    def run_jx_query(self, query, user):
+        data = json.dumps(query, ensure_ascii=False)
+        result = requests.post(self.configuration['host_url']+"/query", data=data)
+        response = json.loads(result.content)
+
+        if response.get('type') == "ERROR":
+            cause = find_cause(response)
+            raise Exception(cause)
+        return response
+
+    def run_query(self, annotated_query, user):
+        request = {}
+        comment, request["sql"] = annotated_query.split("*/", 2)
+        meta = request['meta'] ={}
+        for kv in comment.strip()[2:].split(","):
+            k, v = map(unicode.strip, kv.split(":"))
+            meta[k] = v
+
+        logger.debug("Send ActiveData a SQL query: %s", request['sql'])
+        data = json.dumps(request, ensure_ascii=False)
+        result = requests.post(self.configuration['host_url']+"/sql", data=data)
+        response = json.loads(result.content)
+
+        if response.get('type') == "ERROR":
+            cause = find_cause(response)
+            return None, cause
+
+        output = normalize(response)
+        json_data = json.dumps(output, cls=JSONEncoder)
+        return json_data, None
+
+
+
+def normalize(table):
+    columns = {}  # MAP FROM name TO (MAP FROM type TO (full_name))
+    output = []
+
+    def get_unique_name(name, type):
+        all_types = columns.get(name)
+        if all_types is None:
+            all_types = columns[name] = {}
+        specific_type = all_types.get(type)
+        if specific_type is None:
+            if all_types:
+                specific_type = all_types[type] = name + "." + type
+            else:
+                specific_type = all_types[type] = name
+        return specific_type
+
+    for r in table['data']:
+        new_row = {}
+        for i, cname in enumerate(table['header']):
+            val = r[i]
+            if val == None:
+                continue
+            type_ = val.__class__
+            if type_ in [dict, list]:
+                val = json.dumps(val, cls=JSONEncoder)
+            col = get_unique_name(cname, types_map[type_])
+            new_row[col] = val
+        output.append(new_row)
+
+    output_columns = [
+        {
+            "name": full_name,
+            "type": ctype,
+            "friendly_name": full_name
+        }
+        for cname, types in columns.items()
+        for ctype, full_name in types.items()
+    ]
+
+    return {
+        'columns': output_columns,
+        'rows': output
+    }
+
+
+def find_cause(e):
+    while e.get('cause') is not None:
+        c = e['cause']
+        if isinstance(c, list):
+            e = c[0]
+        else:
+            e = c
+    return e.get('template')
+
+register(ActiveData)
```

Comment:
I do not see them, but I am blind to formatting issues.  Maybe we should run a pep8 auto-formater over it.